tests/integ/test_column_names.py

```diff
@@ -254,49 +297,66 @@ def test_rank_related_function_expression(session):
         first_value(df1[" a"]).over(window),
         last_value(df1[" a"]).over(window),
     )
-    assert (
-        [x.name for x in df2._output]
-        == df2.columns
-        == get_metadata_names(session, df2)
-        == [
+
+    verify_column_result(
+        session,
+        df2,
+        [
             '"LAG("" A"", 1, NULL) OVER (  ORDER BY "" A"" ASC NULLS FIRST )"',
             '"LEAD("" A"", 1, NULL) OVER (  ORDER BY "" A"" ASC NULLS FIRST )"',
             '"FIRST_VALUE("" A"") OVER (  ORDER BY "" A"" ASC NULLS FIRST )"',
             '"LAST_VALUE("" A"") OVER (  ORDER BY "" A"" ASC NULLS FIRST )"',
-        ]
+        ],
+        [StringType(), StringType(), StringType(), StringType()],
+        [Row(None, None, "v", "v")],
     )
+
     df3 = df1.select(
         lag(df1["a"]).over(window),
         lead(df1["a"]).over(window),
         first_value(df1["a"]).over(window),
         last_value(df1["a"]).over(window),
     )
-    assert (
-        [x.name for x in df3._output]
-        == df3.columns
-        == get_metadata_names(session, df3)
-        == [
+
+    verify_column_result(
+        session,
+        df3,
+        [
             '"LAG(""A"", 1, NULL) OVER (  ORDER BY "" A"" ASC NULLS FIRST )"',
             '"LEAD(""A"", 1, NULL) OVER (  ORDER BY "" A"" ASC NULLS FIRST )"',
             '"FIRST_VALUE(""A"") OVER (  ORDER BY "" A"" ASC NULLS FIRST )"',
             '"LAST_VALUE(""A"") OVER (  ORDER BY "" A"" ASC NULLS FIRST )"',
-        ]
+        ],
+        [
+            LongType() if not local_testing_mode else StringType(),
+            LongType() if not local_testing_mode else StringType(),
+            LongType(),
+            LongType(),
+        ],
+        [Row(None, None, 1, 1)],
     )
 
 
-def test_literal(session):
+def test_literal(session, local_testing_mode):
     df1 = session.create_dataframe([[]])
     df2 = df1.select(lit("a"), lit(1), lit(True), lit([1]))
-    assert (
-        [x.name for x in df2._output]
-        == df2.columns
-        == get_metadata_names(session, df2)
-        == [
+
+    expected_dtypes = (
+        [StringType(), LongType(), BooleanType(), ArrayType(LongType())]
+        if local_testing_mode
+        else [StringType(), LongType(), BooleanType(), ArrayType(StringType())]
```

Comment:
@sfc-gh-jdu for this case,  it actually seems should be ArrayType(LongType), but without local testing mode, we seems getting ArrayType(StringType()), do you know if that is what we expected today?

Reply:
Yes, because inner type of an array is not enforced, because it's not a structured type. So the server will just return StringType

Reply:
I see, so that is an expected behavior with snowpark on snowflake then.